### PR TITLE
Apply porous media changes

### PR DESCRIPTION
**Pull request by yifan-flex**

### CODE REVIEW

File: flow360/component/constants.py [hunk 1]
```diff
@@ -3,7 +3,13 @@
 """
 
 
-class constants:
+class PhysicalConstants:
     """physical constants"""
 
     R = 287.0529
```
Comment (maciej-flexcompute) [inline flow360/component/constants.py:9]: It seems this `R` is not in use, we can remove
R is defined here: https://github.com/flexcompute/Flow360/blob/develop/flow360/component/flow360_params/physical_properties.py

File: flow360/component/flow360_params/flow360_params.py [hunk 2]
```diff
@@ -926,19 +928,51 @@ def check_bet_disks_3d_coefficients_in_polars(cls, values):
...
-    windowing_lengths: Optional[Coordinate] = pd.Field(alias="windowingLengths")
+    lengths: Size = pd.Field()
+    windowing_lengths: Optional[Size] = pd.Field(alias="windowingLengths")
+
+
+class PorousMediumVolumeZoneMesh(PorousMediumVolumeZone):
+    """:class:`PorousMediumVolumeZoneMesh` class"""
+
+    zone_type: Literal["mesh"] = pd.Field("mesh", alias="zoneType", const=True)
+    zone_name: str = pd.Field(alias="zoneName")
```
Comment (maciej-flexcompute) [inline flow360/component/flow360_params/flow360_params.py:964]: do we need validator to check if zone of `zone_name` exist on volume zone list?

File: flow360/component/flow360_params/flow360_params.py [hunk 3]
```diff
@@ -926,19 +928,51 @@ def check_bet_disks_3d_coefficients_in_polars(cls, values):
...
+    zone_type: Literal["box"] = pd.Field("box", alias="zoneType", const=True)
     center: Coordinate = pd.Field()
-    lengths: Coordinate = pd.Field()
-    axes: List[Coordinate] = pd.Field(min_items=2, max_items=3)
-    windowing_lengths: Optional[Coordinate] = pd.Field(alias="windowingLengths")
+    lengths: Size = pd.Field()
+    windowing_lengths: Optional[Size] = pd.Field(alias="windowingLengths")
+
+
+class PorousMediumVolumeZoneMesh(PorousMediumVolumeZone):
```
Comment (maciej-flexcompute) [inline flow360/component/flow360_params/flow360_params.py:960]: Is this 23.3.2.0 or 24.1 solver version?

File: flow360/component/flow360_params/flow360_params.py [hunk 4]
```diff
@@ -943,8 +944,22 @@ def check_bet_disks_3d_coefficients_in_polars(cls, values):
         return _check_bet_disks_3d_coefficients_in_polars(values)
 
 
-class PorousMediumVolumeZone(Flow360BaseModel):
-    """:class:`PorousMediumVolumeZone` class"""
+# pylint: disable=too-few-public-methods
+class PorousMedium(PorousMediumVolumeZone):
```
Comment (chai-flex) [inline flow360/component/flow360_params/flow360_params.py:948]: Why does PorousMedium extend PorousMediumVolumeZone?

Should we have a base abstract class instead that is inherited by both the above classes?

File: flow360/component/flow360_params/unit_system.py [hunk 5]
```diff
@@ -827,6 +901,41 @@ class Flow360HeatFluxUnit(_Flow360BaseUnit):
...
+
+
+class Flow360ThermalConductivityUnit(_Flow360BaseUnit):
+    """:class: Flow360ThermalConductivityUnit"""
+
+    dimension_type = ThermalConductivityType
+    unit_name = "flow360_thermal_conductivity_unit"
+
+
+class Flow360InverseAreaUnit(_Flow360BaseUnit):
```
Comment (chai-flex) [inline flow360/component/flow360_params/unit_system.py:925]: Why do we need extra inverse area and length. Aren't these defined by other quantities?

File: tests/test_unit_system.py [hunk 6]
```diff
@@ -484,6 +484,13 @@ def test_unit_system_init():
         "moment": {"value": 1.0, "units": "N*m"},
         "angular_velocity": {"value": 1.0, "units": "rad/s"},
         "heat_flux": {"value": 1.0, "units": "kg/s**3"},
+        "heat_source": {"value": 1.0, "units": "kg/s**3/m"},
```
Comment (chai-flex) [inline tests/test_unit_system.py:487]: What is the purpose of defining such units, is it for ensuring that the user is providing the right units when specifying these terms? is the unit system smart enough to do the MLTK dimensional analysis?

**Review by chai-flex — APPROVED** [review]

**Pull request merged**